### ISSUE TITLE
Strange hbond analysis deprecation

### ISSUE THREAD

**Issue by kain88-de** [MDAnalysis/mdanalysis#1668]
### Expected behaviour

Using the hbond analysis doesn't raise any warnings

### Actual behaviour

Our own test code raises the about 150+ deprecation warnings for hbonds.  

`_get_bonded_hydrogens_list()` is deprecated but called internally by the hbond analysis all the time.

https://github.com/MDAnalysis/mdanalysis/blob/2946aa410afb7c250af5015dfd5fa25dc989a8a0/package/MDAnalysis/analysis/hbonds/hbond_analysis.py#L795

I'm not sure how to resolve this since it tells the user to initiate the class differently then to call this function. The code was written by @orbeckst in 2011. So does this still apply?

### Currently version of MDAnalysis:
(run `python -c "import MDAnalysis as mda; print(mda.__version__)"`)
0.17.dev

**Comment by orbeckst** [comment]
Is it really called "all the time"?
> _get_bonded_hydrogens_list() is deprecated but called internally by the hbond analysis all the time.

because the default is 'distance'.  'heuristic' can be chosen to the user if they know what they are doing. 

To be honest, I am not sure if we need to keep the 'heuristic' version around. We could simply retire it. I don't remember any benchmarking results that compared the speed difference between then two but one could argue that we rather be always correct than fast but only sometimes correct.

**Comment by kain88-de** [comment]
OK I misunderstood the deprecation warning, because it refers to a function name and a setting chosen in `init`. Thanks for clearing that up. This means for one we should update the warning message to make it clearer to the user and us. Also since this method is deprecated for years now. When do we want to remove it? Do we still want to remove it?

**Comment by orbeckst** [comment]
I just checked `git blame` and apparently I added this warning in 2012... I don't think that we even had a clear idea how we would deprecate features then. `DeprecationWarning` seems the wrong classification. 

However, now I suggest to **deprecate it for 1.0** – no-one should be using it but it will users time to react. (We can then remove the `detect_hydrogens` kwarg and the whole machinery to choose an algorithm.)
